qiskit_optimization/applications/clique.py

```diff
@@ -0,0 +1,120 @@
+                 size: Optional[int] = None) -> None:
+        """
+        Args:
+            graph: A graph representing a clique problem. It can be specified directly as a
+            NetworkX Graph, or as an array or list if format suitable to build out a NetworkX graph.
+            size: The size of the clique
+        """
+        super().__init__(graph)
+        self._size = size
+
+    def to_quadratic_program(self) -> QuadraticProgram:
+        """Convert a clique problem instance into a
+        :class:`~qiskit_optimization.problems.QuadraticProgram`
+
+        Returns:
+            The :class:`~qiskit_optimization.problems.QuadraticProgram` created
+            from the clique problem instance.
+        """
+        complement_g = nx.complement(self._graph)
+
+        mdl = Model(name='Clique')
+        n = self._graph.number_of_nodes()
+        x = {i: mdl.binary_var(name='x_{0}'.format(i)) for i in range(n)}
+        for w, v in complement_g.edges:
+            mdl.add_constraint(x[w] + x[v] <= 1)
+        if self.size is None:
+            mdl.maximize(mdl.sum(x[i] for i in x))
+        else:
+            mdl.add_constraint(mdl.sum(x[i] for i in x) == self.size)
+        op = QuadraticProgram()
+        op.from_docplex(mdl)
+        return op
+
+    def interpret(self, result: Union[OptimizationResult, np.ndarray]) -> List[int]:
+        """Interpret a result as a list of node indices
+
+        Args:
+            result : The calculated result of the problem
+
+        Returns:
+            The list of node indices whose corresponding variable is 1
+        """
+        x = self._result_to_x(result)
+        clique = []
+        for i, value in enumerate(x):
+            if value:
+                clique.append(i)
+        return clique
+
+    def draw(self, result: Optional[Union[OptimizationResult, np.ndarray]] = None,
+             pos: Optional[Dict[int, np.ndarray]] = None) -> None:
+        """Draw a graph with the result. When the result is None, draw an original graph without
+        colors.
+
+        Args:
+            result : The calculated result for the problem
+            pos: The positions of nodes
+        """
+        if result is None:
+            nx.draw(self._graph, pos=pos, with_labels=True)
+        else:
+            x = self._result_to_x(result)
+            nx.draw(self._graph, node_color=self._node_colors(x), pos=pos, with_labels=True)
+
+    def _node_colors(self, x: np.ndarray) -> List[str]:
+        # Return a list of strings for draw.
+        # Color a node with red when the corresponding variable is 1.
+        # Otherwise color it with dark gray.
+        return ['r' if value else 'darkgrey' for value in x]
+
+    @property
+    def size(self) -> int:
+        """Getter of size
+
+        Returns:
+            The size of the clique
+        """
+        return self._size
+
+    @size.setter
+    def size(self, size: int) -> None:
```

Comment:
It was difficult to understand without comments. I added some comments. Changing the value of `size` does not affect interpreting a result or drawing the result. When "size" is None, `to_quadratic_probgram` makes an optimization model for a maximal clique instead of the specified size of a clique.